### ISSUE TITLE
For optimal immersion you should not see the Windows Desktop environment when transisting between RetroArch to Core and back again

### ISSUE THREAD

**Issue by porg** [libretro/RetroArch#13647]
## Reproduction
1. Open RetroArch in fullscreen mode. Great immersion!
2. Open a playlist, select a game, launch it.
3. Until the core is loaded, you shortly see the Windows Desktop environment. (few secs)
4. You play the game, in fullscreen mode. Great immersion!
5. Exit the game.
6. Until RetroArch is loaded again, you shortly see the Windows Desktop environment. (few secs)

## Improvement Proposal
- You do not see the Windows Desktop environment during the visual transitions.
  - Neither when launching content (2->3) nor when exiting a core (5->6)
  - Because this destroys the immersive gaming experience.
- Instead you visually directly transist from RetroArch to Core (with the nice game info launch screen, compliments!) and from the Core back to RetroArch.
- On my Chromebook running ChromeOS, it works exactly that way!
- Would be great to have that experience on Windows too.

## Workaround
- As step 0 launch a fullscreen browser or image app with a nice visual.
- Then you will see this in the short Desktop interludes.
- But this also does not work out entirely! Because when transisting through Microsoft Edge in fullscreen mode the Windows menu bar shortly shows and hides. So even that workaround cannot avoid the disruptive experience fully.

## Version/Commit
- RetroArch: 1.9.1 git e954a46 built on 2021-12-06
- OS: Windows 10 Home - 21H1 (Build 19043.1526)

**Comment by porg** [comment]
What else?
- Settings > User Interface > Disable Desktop Composition
  - Sounded that that could do what I intended to achieve here.
  - Changed it from default OFF to ON. But it did not do any noticeable change. And no change in the spirit of this issue.
- Read [anecdotal evidence](https://www.reddit.com/r/RetroArch/comments/i7579d/retroarch_shows_windows_desktop_when_loading_a_rom/) "That this was always the case with RetroArch on Windows. And was already once proposed. But closed as Wontfix."

**Comment by hizzlekizzle** [comment]
Yeah, it turns out this is much more complicated than it seems. The reason you see the desktop is because RetroArch has to completely tear down the existing context to be able to load the next one. Someone tried a few months ago to allow cores with compatible contexts to skip the teardown (e.g., OpenGL -> OpenGL), but it ended up being a big mess with a lot of edge cases and pitfalls (IIRC)--and it frequently had to tear down anyway--so they eventually abandoned it.

**Comment by jayare5** [comment]
This is actually an annoying issue with Lakka too!
You see the Lakka flower in between loading core and menu.
Can RetroArch load a secondary instance with just black covering the screen?  Actually, I understand that even this might be awkward too if using RA inside a Desktop OS as sometimes full-screen apps appear behind the task bar. 

Wow, this does sound like quite a challenge.

**Comment by macsgv** [comment]
The same thing happens to me using Windows 11, as front-end I use ArcadeFlow and for retroarch emulators.

When I load a game from arcadeflow it calls retroarch's MAME core via CLI. So far everything is perfect :-)

But the transition from loading the game until it is seen on the screen is not clean, there is a flicker of less than 1 second that shows the Windows 11 taskbar, so far I have not found a solution to make the transition smooth totally black screen.

**Comment by sempiternoiddqd** [comment]
I created a launcher to solve this problem, in my profile, my repo: **RetroBlackScreen**

https://github.com/sempiternoiddqd/RetroBlackScreen
